### PR TITLE
SecurityPolicyService.IsSubscribed should ignore state

### PR DESCRIPTION
**Pull request by xavierdecoster**
Fixes NuGet/Engineering#1970

### CODE REVIEW

File: src/NuGetGallery/Security/SecurityPolicyService.cs [hunk 1]
```diff
@@ -354,7 +354,10 @@ public bool IsSubscribed(User user, IUserSecurityPolicySubscription subscription
             var subscribed = FindPolicies(user, subscription);
             var required = subscription.Policies;
 
-            return required.All(rp => subscribed.Any(sp => sp.Equals(rp)));
+            return required.All(rp => 
+                    subscribed.Any(sp => 
+                        string.Equals(sp.Name, rp.Name, StringComparison.OrdinalIgnoreCase)
```
Comment (skofman1) [inline src/NuGetGallery/Security/SecurityPolicyService.cs]: unit tests?

Comment (xavierdecoster) [reply src/NuGetGallery/Security/SecurityPolicyService.cs]: Actually, just found a shortcut to cover this particular case: https://github.com/NuGet/NuGetGallery/pull/6753/commits/326b80d769b134cb1aee022ee9f9d4ed24dd91f2

File: tests/NuGetGallery.Facts/Areas/Admin/Controllers/SecurityPolicyControllerFacts.cs [hunk 2]
```diff
@@ -188,6 +188,30 @@ public async Task UpdateUnsubscribesUsers()
...
+            policyService.MockEntitiesContext.ResetCalls();
+
+            // Act.
+            // Simulates changes to the configurable state of all existing policy subscriptions
+            users.ForEach(u => 
+                u.SecurityPolicies.Where(p => p.Subscription == subscription.SubscriptionName).ToList().ForEach(p => 
+                    p.Value = Guid.NewGuid().ToString()));
+
+            // Assert.
+            users.ForEach(u => Assert.True(policyService.IsSubscribed(u, subscription)));
```
Comment (joelverhagen) [inline tests/NuGetGallery.Facts/Areas/Admin/Controllers/SecurityPolicyControllerFacts.cs]: `Assert.All` provides a bit better of a message

Comment (xavierdecoster) [reply tests/NuGetGallery.Facts/Areas/Admin/Controllers/SecurityPolicyControllerFacts.cs]: Addressed with https://github.com/NuGet/NuGetGallery/pull/6753/commits/7b002ae8b7accac8601607e3ef17ed5dc5fa6d82

**Pull request merged**